components/gitpod-cli/cmd/info.go

```diff
@@ -0,0 +1,94 @@
+// Copyright (c) 2022 Gitpod GmbH. All rights reserved.
+// Licensed under the GNU Affero General Public License (AGPL).
+// See License-AGPL.txt in the project root for license information.
+
+package cmd
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/gitpod-io/gitpod/common-go/log"
+	supervisor_helper "github.com/gitpod-io/gitpod/gitpod-cli/pkg/supervisor-helper"
+	supervisor "github.com/gitpod-io/gitpod/supervisor/api"
+	"github.com/olekukonko/tablewriter"
+	"github.com/spf13/cobra"
+	"google.golang.org/grpc"
+	"os"
+	"time"
+)
+
+var infoCmdOpts struct {
+	// Json configures whether the command output is printed as JSON, to make it machine-readable.
+	Json bool
+}
+
+// infoCmd represents the info command.
+var infoCmd = &cobra.Command{
+	Use:   "info",
+	Short: "Display workspace info, such as its ID, class, etc.",
+	Run: func(cmd *cobra.Command, args []string) {
+		ctx, cancel := context.WithTimeout(cmd.Context(), 5*time.Second)
+		defer cancel()
+
+		conn, err := supervisor_helper.Dial(ctx)
+		if err != nil {
+			log.Fatal(err)
+		}
+		defer conn.Close()
+
+		data, err := fetchInfo(ctx, conn)
+		if err != nil {
+			log.Fatal(err)
+		}
+
+		if infoCmdOpts.Json {
+			content, _ := json.Marshal(data)
+			fmt.Println(string(content))
+			return
+		}
+
+		outputInfo(data)
+	},
+}
+
+func fetchInfo(ctx context.Context, conn *grpc.ClientConn) (*infoData, error) {
+	wsInfo, err := supervisor.NewInfoServiceClient(conn).WorkspaceInfo(ctx, &supervisor.WorkspaceInfoRequest{})
+	if err != nil {
+		return nil, fmt.Errorf("failed to retrieve workspace info: %w", err)
+	}
+
+	return &infoData{
+		WorkspaceId:    wsInfo.WorkspaceId,
+		InstanceId:     wsInfo.InstanceId,
+		WorkspaceClass: wsInfo.WorkspaceClass,
+		WorkspaceUrl:   wsInfo.WorkspaceUrl,
+		ClusterHost:    wsInfo.WorkspaceClusterHost,
+	}, nil
+}
+
+func outputInfo(info *infoData) {
```

Comment:
I liked this style very much, I think we should consider it for `gp top` as well!

Reply:
gave it a go, what do you think of this:
![image](https://user-images.githubusercontent.com/9721064/194033759-0aecedbf-994a-48fd-a3da-03149b818883.png)

Reply:
when we extended the `top` cmd to include the workspace class, I wasn't convinced by the layout of mixing table + plain text, it would be nice to make class feel more part of the output. However, now we also have `gp info` so maybe we don't need class in `gp top`? I think if you open a draft PR we can collect feedback in there :)

Reply:
good idea of moving the class to the table, raised a draft PR here: https://github.com/gitpod-io/gitpod/pull/13607